orchestrator/forms/network_type_validators.py

```diff
@@ -12,54 +12,27 @@
 # limitations under the License.
 
 
-from typing import Any, Dict, Iterator, Optional
+from typing import Optional
 
-from pydantic import BaseModel, ConstrainedInt, conint, root_validator
-
-from orchestrator.utils.vlans import VlanRanges
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from typing_extensions import Annotated
 
 
 class BFD(BaseModel):
-    class Config:
-        schema_extra = {"format": "optGroup"}
+    model_config = ConfigDict(json_schema_extra={"format": "optGroup"})
 
     # order matters, this should be first
     enabled: bool
-    minimum_interval: Optional[conint(ge=1, le=255000)] = 900  # type: ignore
-    multiplier: Optional[conint(ge=1, le=255)] = 3  # type: ignore
-
-    @root_validator()
-    def check_optional_fields(cls, values: Dict) -> Dict:  # noqa: B902
-        if not values.get("enabled"):
-            values.pop("minimum_interval", None)
-            values.pop("multiplier", None)
-
-        return values
-
-
-class MTU(ConstrainedInt):
-    ge = 1500
-    le = 9000
+    minimum_interval: Optional[Annotated[int, Field(ge=1, le=255000)]] = 900
+    multiplier: Optional[Annotated[int, Field(ge=1, le=255)]] = 3
 
-    @classmethod
-    def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
-        super().__modify_schema__(field_schema)
-        field_schema["multipleOf"] = 7500
+    @model_validator(mode="after")
+    def check_optional_fields(self) -> "BFD":
+        if not self.enabled:
+            self.minimum_interval = None
+            self.multiplier = None
 
+        return self
 
-class VlanRangesValidator(VlanRanges):
```

Comment:
what happened to this validator?

Reply:
Added validation to the VlanRanges type in nwa-stdlib. But I see now that we need custom validation/json schema sometimes, for Nsi VlanRanges etc.

I didn't think this through..

Will try to add the VlanRangesValidator back here and make it wrap VlanRanges, couldn't get it to work though..